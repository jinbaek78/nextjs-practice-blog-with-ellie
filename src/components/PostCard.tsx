import { Post } from '@/service/post';
import Image from 'next/image';

type Props = {
  postData: Post;
};
export default function PostCard({
  postData: { path, date, title, description, category },
}: Props) {
  console.log(path, date, title);
  return (
    <li className="flex flex-col gap-1 shadow-xl rounded-lg">
      <Image
        className="rounded-lg"
        src={`/images/posts/${path}.png`}
        alt="post-image"
        width={400}
        height={150}
      />
      <p className="text-right text-zinc-800 mr-3">{date}</p>
      <div className="flex flex-col justify-center items-center">
        <h3 className="font-semibold text-lg">{title}</h3>
        <h5 className="text-sm">{description}</h5>
        <div className="bg-green-200 rounded-xl py-1 px-4 text-sm mt-3 mb-5">
          {category}
        </div>
      </div>
    </li>
  );
}
