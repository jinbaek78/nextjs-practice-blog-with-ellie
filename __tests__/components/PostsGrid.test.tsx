import postsData from '../../public/data/posts.json';
import { render, screen } from '@testing-library/react';
import PostCard from '@/components/PostCard';
import PostsGrid from '@/components/PostsGrid';
jest.mock('@/components/PostCard');

describe.skip('PostGrid', () => {
  const posts = postsData;
  afterEach(() => {
    (PostCard as jest.MockedFunction<typeof PostCard>).mockReset();
  });
  it('renders with posts correctly', () => {
    render(<PostsGrid posts={posts} />);
    expect(screen.getAllByRole('listitem')).toHaveLength(posts.length);
  });

  it('renders correctly', () => {
    const { asFragment } = render(<PostsGrid posts={posts} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
