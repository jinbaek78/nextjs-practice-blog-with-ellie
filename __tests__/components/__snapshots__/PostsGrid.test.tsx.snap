// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostGrid renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="grid gap-4 grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4"
  >
    <li />
    <li />
    <li />
    <li />
    <li />
    <li />
    <li />
    <li />
    <li />
  </ul>
</DocumentFragment>
`;
